tests/requests/tests.py

```diff
@@ -669,6 +670,20 @@ def read(self, len=0):
         with self.assertRaises(UnreadablePostError):
             request.FILES
 
+    def test_pickle_request(self):
+        request = HttpRequest()
+        request.method = "GET"
+        request.META["QUERY_STRING"] = ";some=query&+query=string"
+        request.COOKIES = {"post-key": "post-value"}
+        dump = pickle.dumps(request)
+        request_from_pickle = pickle.loads(dump)
+        with self.assertRaises(AttributeError) as cm:
+            self.assertEqual(repr(request), repr(request_from_pickle))
+        self.assertEqual(
+            str(cm.exception),
+            "'HttpRequest' object has no attribute 'META'",
+        )
```

Comment:
We should also add tests for actual requests (`WSGIRequest`, `ASGIRequest`) which crash when removing any key from `non_picklable_attrs`.